sequencer/src/api.rs

```diff
@@ -425,25 +425,26 @@ pub mod test_helpers {
         }
     }
 
-    impl<const NUM_NODES: usize, P, C> TestNetworkConfigBuilder<{ NUM_NODES }, P, C>
-    where
-        P: PersistenceOptions,
-        C: StateCatchup + 'static,
+    impl<const NUM_NODES: usize>
+        TestNetworkConfigBuilder<{ NUM_NODES }, no_storage::Options, MockStateCatchup>
     {
-        /// NOTICE: This function will reset most of the config. Please assign the number of the nodes
-        /// before assigning values.
-        pub fn num_nodes<const N: usize>(
-            self,
+        pub fn with_num_nodes<const N: usize>(
```

Comment:
Nit: we don't need this `N` parameter anymore because we can take it from `NUM_NODES`, and then you can do `TestNetworkConfigBuilder::<WhateverNumberOfNodes>::with_num_nodes()`. Actually we could just change `with_num_nodes` to `new` at that point lol. But this is minor